graph/patterns/dictionary.md

```diff
@@ -1,26 +1,57 @@
-# Dictionary types
+# Dictionary
+
+Microsoft Graph API Design Pattern
+
+*The Dictionary type provides the ability to create a set of primitives or objects, of the same type, where the API consumer can define a name for each value in the set.*
+
+## Problem
+--------
+
+The API design requires a resource to include an unknown quantity of data elements of the same type that must be named using values provided by the API consumer.
+
+## Solution
+--------
+
+API designers use a JSON object to represent a dictionary in a `application/json`response payload. When describing the model in CSDL, a new complex type can be created that derives from `Org.OData.Core.V1.Dictionary` and then use the `Org.OData.Validation.V1.OpenPropertyTypeConstraint`to constrain the type that can be used for the values in the dictionary.
+
+Dictionary entries can be added via `POST`, updated via `PATCH`, and they can be removed by setting the entry value to `null`. Multiple entries can be updated at once by using `PATCH` on the dictionary property.
+
+## Issues and Considerations
+-------------------------
 
 Dictionaries, sometimes called maps, are a collection of name-value pairs. They allow dynamic data sets to be accessed in a systematic manner and are a good compromise between a strictly defined ahead of time structure with all its named properties and between a loosely defined dynamic object (i.e. OData OpenTypes).
 
+As dictionary entries are removed via setting the value to null, this means that dictionaries can only support values that are non-nullable.
+
+  OpenQuestions:
+  - Can/should PUT be supported on the dictionary property and/or the entry value 
+  - What does OData say about being able to POST to a structured property? Will OData Web API allow that?
+  - Must an implementer support PATCH at both the dictionary level and the entry level?
+
 More information:
 
 - [OData reference](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#dictionary)
 
-## When to use dictionary types
+ 
+## When to Use this Pattern
+------------------------
 
 Before using a dictionary type in your API definition make sure your scenario fits the following criteria:
 
 - The data values MUST be related to one another semantically as a collection.
 - The value types MUST be a primitive type or is a **ComplexType**. Mixed primitive types are not allowed.
-- The client MUST define the keys of this type. As opposed to the service defining it in advance.
+- The client MUST define the keys of this type. As opposed to the service defining them in advance.
```

Comment:
I wonder if this is clear enough
I wonder if we need another statement to be explicit what this is NOT for.  The "As opposed" statement doesn't go far enough for me

"Services MUST NOT use dictionary types as a way to quickly iterate APIs without strong typing or API governance"

Reply:
Let's see if people try to abuse them.  I'd rather not starting looking for all thing things that folks might do wrong and enumerating them.